src/IndexNameMapper.php

```diff
@@ -44,6 +48,9 @@ public function getIndexNameFromClass(string $className): string
         return $this->getPrefixedIndex($indexName);
     }
 
+    /**
+     * @throws ExceptionInterface
+     */
     public function getClassFromIndexName(string $indexName): string
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏Does PHPStan **really** need this PHPDoc? 

I mean, there is no hidden code here, the method is clear about which Exception can occurs.

For example LEVEL 9 does not trigger any warning here: https://phpstan.org/r/6a1d0c05-47f0-42ec-bae9-acca046c4908

Reply:
The playground is not a good proof, because the exception analysis is not enabled on it.

PHPStan is only looking at the code inside the function when he's doing the analysis of this function. But when analysis is done, he's only looking at phpdoc like the method was coming from an interface. Especially because someone could extend the class and override the method implementation.

In the same way, no error is reported on the `test` method for https://phpstan.org/r/2fbe6b49-15e6-473a-b9f3-aceb4676abdd.

So yes, PHPStan really need the PHPDoc (and in a general way, always need the phpdoc on interface or methods)